src/execute.c

```diff
@@ -687,6 +687,14 @@ jv jq_next(jq_state *jq) {
         set_error(jq, jv_invalid_with_msg(msg));
         goto do_backtrack;
       }
+      // $array | .[-1]
+      if (jv_get_kind(k) == JV_KIND_NUMBER && jv_get_kind(t) == JV_KIND_ARRAY) {
+        int idx = jv_number_value(k);
+        if (idx < 0) {
+          jv_free(k);
+          k = jv_number(jv_array_length(jv_copy(t)) + idx);
+        }
+      }
```

Comment:
This patch should be authored by @nicowilliams

Reply:
@emanuele6 - It was.